src/blob/persistence/LokiBlobMetadataStore.ts

```diff
@@ -1343,22 +1619,37 @@ export default class LokiBlobMetadataStore
       context
     );
 
+    if (isVersionProvided) {
+      // TODO: Verify production azure behaviour when specifying snapshots to delete.
+      coll.findAndRemove({
+        accountName: account,
+        containerName: container,
+        name: blob,
+        snapshot: "",
+        versionId: versionId
+      });
+      return;
+    }
+
     // Scenario: Delete base blob only
     if (againstBaseBlob && options.deleteSnapshots === undefined) {
       const count = coll.count({
         accountName: account,
         containerName: container,
-        name: blob
+        name: blob,
+        snapshot: { $gt: "" } // Only count actual snapshots, not empty snapshot (base blob)
       });
-      if (count > 1) {
+      if (count > 0) {
         throw StorageErrorFactory.getSnapshotsPresent(context.contextId!);
       } else {
-        coll.findAndRemove({
-          accountName: account,
-          containerName: container,
-          name: blob
-        });
+        if (this.isBlobVersioningEnabled()) {
+          doc.isCurrentVersion = false;
+          coll.update(doc);
+        } else {
+          coll.remove(doc);
+        }
```

Comment:
Check the delete on storage account without softdelete on real azure , and make sure they are aligned.

Reply:
If you delete a specific version, it is just deleted. 

Versioned base blobs are always set to previous version with or without versioning disabled. Added a comment for this